src/server/lib/app/lobby.ex

```diff
@@ -0,0 +1,102 @@
+defmodule App.Lobby do
+  use GenServer
+
+  #Starts a new lobby with the given parameters.
+  def create_lobby(name, params \\ %{}) do
+    defaults = %{
+      target_guess_range: "local landmarks",
+      time_per_round: 15,
+      number_of_rounds: 1
+    }
+
+    state = Map.merge(defaults, params)
+    GenServer.start_link(__MODULE__, %{name: name, state: state, users: %{}, round_data: []}, name: via_tuple(name))
+  end
+
+  #Returns information about the lobby if the calling user is associated with it.
+  def get_lobby_info(name, user_pid) do
+    GenServer.call(via_tuple(name), {:get_lobby_info, user_pid})
+  end
+
+  #Marks the calling user as ready. If all users are ready, starts the round countdown.
+  def readyup(name, user_pid) do
+    GenServer.call(via_tuple(name), {:readyup, user_pid})
+  end
+
+  #Allows a user to join an existing lobby by name.
+  def join(name, user_pid) do
+    GenServer.call(via_tuple(name), {:join, user_pid})
+  end
+
+  #Submits a user's guess for the current round.
+  def submit_lobby_guess(name, user_pid, guess_data) do
+    GenServer.call(via_tuple(name), {:submit_guess, user_pid, guess_data})
+  end
```

Comment:
these are flipped in my stylistic view. these functions should be implementing the just "business" logic, where as the `handle_call`'s should be implementing the actor logic only.